src/shared/debug/launchConfiguration.ts

```diff
@@ -84,3 +94,60 @@ class DefaultDebugConfigSource implements DebugConfigurationSource {
         await this.launch.update('configurations', value)
     }
 }
+
+function getExistingSamTemplateTargets(launchConfig: LaunchConfiguration): TemplateTargetProperties[] {
```

Comment:
nit: "existing" in getExisting*Targets is superfluous. You have a launch config, and you want the config entries related to a certain target type. "get*Targets" or "get*Configs" is just as meaningful with less verbosity.